src/LinuxTracing/LeafFunctionCallManager.cpp

```diff
@@ -46,14 +47,32 @@ Callstack::CallstackType LeafFunctionCallManager::PatchCallerOfLeafFunction(
   ORBIT_CHECK(unwinder != nullptr);
   ORBIT_CHECK(event_data->GetCallchainSize() > 2);
 
-  uint64_t rbp = event_data->GetRegisters()[PERF_REG_X86_BP];
-  uint64_t rsp = event_data->GetRegisters()[PERF_REG_X86_SP];
+  const uint64_t rbp = event_data->GetRegisters()[PERF_REG_X86_BP];
+  const uint64_t rsp = event_data->GetRegisters()[PERF_REG_X86_SP];
+  const uint64_t ip = event_data->GetRegisters()[PERF_REG_X86_IP];
 
   if (rbp < rsp) {
     return Callstack::kFramePointerUnwindingError;
   }
 
+  // If the frame pointer register is set correctly at the current instruction, there is no need
```

Comment:
I feel like this comment should be on `if (*has_frame_pointer_or_error) {` right below.